### PR TITLE
IM TypeScript

### PR DESCRIPTION
**Pull request by Perfoys**
Converted all JavaScript files to TypeScript files,
Installed TypeScript and Types packages,
Updated app style
Updated redux store

### CODE REVIEW

File: src/components/common/Timer.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { useMemo, FC } from 'react';
+import calculateTimeLeft from '../../helpers/calculateTimeLeft';
+import style from './timer.module.scss';
+
+type TimerProps = {
+  text: string,
+  endDate: Date
+};
+
+const Timer: FC<TimerProps> = ({ text, endDate }: TimerProps) => {
```
Comment (Smolyaninov-Dmitriy) [inline src/components/common/Timer.tsx:10]: should we duplicate TimerProps for props? can we use `const Timer: FC<TimerProps> = ({ text, endDate }) `

File: src/components/modal/StatusSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+type OptionType = {
+  value: string,
+  label: string
+};
+
+const StatusSelector: FC<StatusSelectorProps> = ({ task }: StatusSelectorProps) => {
+  const dispatch = useDispatch();
+  const defaultOption: OptionType = { value: task.status, label: task.status };
+
+  const handleChange = (option: SingleValue<OptionType>) => {
```
Comment (Smolyaninov-Dmitriy) [inline src/components/modal/StatusSelector.tsx:21]: let's wrap this function in useCallback

File: src/components/table/MembersTable.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+const MembersTable: FC<MembersTableProps> = ({ members, tasks }: MembersTableProps) => {
+  const filterTasks = (tasks: Array<ITask>, status: string) => tasks.filter(task => task.status === status);
+
+  return (
+    <Wrapper>
+      <Participants members={members} />
+      <div className={style.table}>
+        <Column
```
Comment (Smolyaninov-Dmitriy) [inline src/components/table/MembersTable.tsx:21]: what do you think about creating a function that returns this component?

File: src/components/common/Timer.tsx (hunk 1 above)
Comment (Perfoys) [reply src/components/common/Timer.tsx]: Sorry, my bad. We can use types once, fixed this problem

File: src/components/modal/StatusSelector.tsx (hunk 2 above)
Comment (Perfoys) [reply src/components/modal/StatusSelector.tsx]: Thanks, solved this

File: src/components/table/MembersTable.tsx (hunk 3 above)
Comment (Perfoys) [reply src/components/table/MembersTable.tsx]: It's a good idea, I created a function for this purpose

**Pull request merged**